src/tasks/task_list.hpp

```diff
@@ -348,22 +351,31 @@ class TaskRegion {
   bool CheckAndUpdate() {
     for (auto &reg_dep : id_for_reg) {
       auto reg_id = reg_dep.first;
-      if (HasRun(reg_id)) {
+      if (id_complete[reg_id]) continue;
+      if (HasRun(reg_id) && !all_done[reg_id].active) {
         bool done = IsComplete(reg_id);
 #ifdef MPI_PARALLEL
-        int all_done = done;
-        int global_done;
-        MPI_Allreduce(&all_done, &global_done, 1, MPI_INT, MPI_MIN, MPI_COMM_WORLD);
+        all_done[reg_id].val = done;
+        all_done[reg_id].StartReduce(MPI_MIN);
+        int global_done = all_done[reg_id].val;
```

Comment:
Is `global_done` no longer needed now that we're checking `all_done[reg_id].active` instead?

Reply:
thanks, cleaning this up.